src/domain/synthetics/markets/useGmMarketsApy.ts

```diff
@@ -44,34 +47,152 @@ function useMarketAddresses(marketsInfoData: MarketsInfoData | undefined) {
+): {
+  [marketAddress: string]: bigint;
+} {
+  const liquidityProvidersIncentives = useLiquidityProvidersIncentives(chainId);
+  const excludeHolders = liquidityProvidersIncentives?.excludeHolders ?? EMPTY_ARRAY;
+  const marketAddresses = useMarketAddresses(marketsInfoData);
+
+  const request = useCallback<
+    () => MulticallRequestConfig<{
+      [marketAddress: string]: {
+        calls: {
+          [holder: string]: {
+            methodName: string;
+            params: any[];
+          };
+        };
+      };
+    }>
+  >(() => {
+    const req: MulticallRequestConfig<{
+      [marketAddress: string]: {
+        calls: {
+          [holder: string]: {
+            methodName: string;
+            params: any[];
+          };
+        };
+      };
+    }> = {};
+
+    for (const marketAddress of marketAddresses) {
+      req[marketAddress] = {
+        abi: TokenAbi.abi,
+        contractAddress: marketAddress,
+        calls: {},
+      };
+
+      for (const holder of excludeHolders) {
+        req[marketAddress].calls[holder] = {
+          methodName: "balanceOf",
+          params: [holder],
+        };
+      }
+    }
+
+    return req;
+  }, [excludeHolders, marketAddresses]);
+
+  const excludedBalancesMulticall = useMulticall(chainId, "useExcludedLiquidityMarketMap", {
+    key: excludeHolders ? [excludeHolders.join(","), marketAddresses.join(",")] : null,
+    request,
+    parseResponse: (res) => {
+      const result: { [marketAddress: string]: bigint } = {};
+
+      for (const marketAddress of marketAddresses) {
+        const marketData = res.data[marketAddress];
+        let totalExcludedBalance = 0n;
+
+        for (const excludedHolder of excludeHolders) {
+          const excludedBalance = BigInt(marketData[excludedHolder]?.returnValues[0] ?? 0);
+          totalExcludedBalance += excludedBalance;
+        }
+
+        result[marketAddress] = totalExcludedBalance;
+      }
+
+      return result;
+    },
+  });
+
+  return excludedBalancesMulticall.data ?? {};
+}
+
 function useIncentivesBonusApr(chainId: number, marketsInfoData: MarketsInfoData | undefined): MarketTokensAPRData {
   const liquidityProvidersIncentives = useLiquidityProvidersIncentives(chainId);
   const { tokensData } = useTokensDataRequest(chainId);
   const marketAddresses = useMarketAddresses(marketsInfoData);
-  const token = liquidityProvidersIncentives ? tokensData?.[liquidityProvidersIncentives.token] : undefined;
+  const { marketTokensData } = useMarketTokensData(chainId, { isDeposit: false });
+  const tokenAddress = liquidityProvidersIncentives?.token;
+  const excludedLiquidityMarketMap = useExcludedLiquidityMarketMap(chainId, marketsInfoData);
 
-  return useMemo(() => {
+  const regularToken = useMemo(() => {
+    if (!tokenAddress || !tokensData) return undefined;
+
+    const lowerCased = tokenAddress.toLowerCase();
+    const key = Object.keys(tokensData).find((key) => key.toLowerCase() === lowerCased);
```

Comment:
not a blocker, could be cleaner to use ethers.utils.getAddress to get checksummed address instead of comparing lower case addresses

there will be no need to iterate over keys